v53 upgrade, vulkan

aszabo314 opened the pull request:
+ net6.0
+ update to Aardvark.Rendering v53, Adaptify 1.1.x
+ remove unnecessary build script
+ switch to Vulkan to get around shader stage 32 sampler limit
+ changed invalid texture formats rgb -> rgba
+ added DevIL
- had to remove displacemntMap.displacementMap in GeometryBuffer.fs because it makes things invisible (unclear why, to be investigated)
- Lightprobe button crashes with [Vulkan] could not get uniform: { ufName = "TranslucencyBias", ufType = Array (8, Float 32, 16) (to be investigated)

aszabo314 committed:
v53 upgrade, vulkan

StefanLeng commented:
Thank you, that is great. 
I will look into the issues, but it may take some time befor I get to it. I am swamped at work at then moment and that is unlikly to change in the next couple  of weeks.
The displacement thing was never working to my satification, maybe I just remove it.

aszabo314 commented:
No stress, we can also look into it when we have the time. Thanks

StefanLeng committed:
simple rendering for light probe

Use only uniforms and  samplers that exists in the light probe render task, otherwise vulcan will crash

StefanLeng commented:
I am finally working on this. Sorry it took so long, the last months where a bit chaotic on my side.

I partially resovled the error with the light probe: While Opengl ingnores missing uniforms if the conditional code path acsessing them is not executed, Vulcan throws an error. The same thing for missing samplers.
 
After I fix that, I can render the light probe, but unfortunaly if I disable it and enable it again or change the position (leading to a new compile of the render task), I get an crash with ERROR: [Vulkan] could not allocate DescriptorSet (-1000069000)

I am still investigating it, without much progress so far.

There are also issues with transparency (wild artefacts) and ambient occlusion (not working at all). I susspect somthing with the gBuffer depth map is the culprit in both cases, need to investigate more.
